terraform/modules/front_door/waf.tf

```diff
@@ -152,4 +157,97 @@ resource "aws_wafv2_web_acl" "this" {
       sampled_requests_enabled   = true
     }
   }
+
+  rule {
+    name     = "login-ip-rate-limit"
+    priority = 7
+
+    action {
+      block {
+        custom_response {
+          response_code = 429
+        }
+      }
+    }
+
+    statement {
+      rate_based_statement {
+        aggregate_key_type = "IP"
+        limit              = 100
+
+        scope_down_statement {
+          regex_pattern_set_reference_statement {
+            arn = aws_wafv2_regex_pattern_set.waf_rate_limit_urls.arn
+
+            field_to_match {
+              uri_path {}
+            }
+
+            text_transformation {
+              priority = 0
+              type     = "URL_DECODE"
+            }
+          }
+        }
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "waf-login-ip-rate-limit"
+      sampled_requests_enabled   = true
+    }
+  }
+
+  rule {
+    name     = "overall-ip-rate-limit"
+    priority = 8
+
+    action {
+      block {
+        custom_response {
+          response_code = 429
+        }
+      }
+    }
+
+    statement {
+      rate_based_statement {
+        aggregate_key_type = "IP"
+        limit              = 2000
+      }
+    }
+
+    visibility_config {
+      cloudwatch_metrics_enabled = true
+      metric_name                = "waf-overall-ip-rate-limit"
+      sampled_requests_enabled   = true
+    }
+  }
+}
+
+resource "aws_wafv2_regex_pattern_set" "waf_rate_limit_urls" {
+  name     = "${var.prefix}-waf-login-url-regex-patterns"
+  provider = aws.us-east-1
+  scope    = "CLOUDFRONT"
+
+  regular_expression {
+    regex_string = "/account/password/new"
+  }
+
+  regular_expression {
+    regex_string = "/account/password/reset-confirmation"
+  }
+
+  regular_expression {
+    regex_string = "/account/sign-in"
+  }
+
+  regular_expression {
+    regex_string = "/account/two-factor-authentication"
+  }
+
+  regular_expression {
+    regex_string = "/account/two-factor-authentication/resend"
+  }

```

Comment:
Looking at `routes.rb` in the app codebase and having a play with the site, I think these are all the login routes that would need a separate "login" rate limit. AFAICT that covers everything, but pointing out as I'm not as familiar with the application.